### ISSUE TITLE
Queue for Sinks causes game to crash

### ISSUE THREAD

**Issue by Timur13240** [peterhaneve/ONIMods#526]
## Expected Behavior

Dupes just should wash their hands as usual, but the game crashes.

## Current Behavior

Dupes go to use restrooms, but after the two of them pass almost simultaneously through the sink - game crashes.

## Game Info
Base game and only _Spaced Out!_ DLC
Build: U53-642443-S

bug also appears on:
Build: U53-642695-S

## Mod Versions

v3.13.0.0
Steam subscription;
Yes, I'm tried to force update mod using _Mod Updater_.

## Steps to Reproduce

1. Dupes uses the toilet.
2. Two of them almost simultaneously passes through one sink.
3. ...
4. PROFIT! The game crashes.

## Bug Demonstration:
https://github.com/user-attachments/assets/c254a84c-99e7-4127-abba-58194da6b716

## Another demo but only with _Queue for Sinks_ mod turned on:
https://github.com/user-attachments/assets/75270304-5e29-4c5f-85b9-1dcd711a5efa

## Output Log (from the 1st video)
```
Exception in: (Minion).IdleChore+States.root.idle.move.MoveTo()
System.TypeLoadException: Could not resolve type with token 01000036 from typeref (expected class 'Worker' in assembly 'Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null')
  at PeterHan.QueueForSinks.WorkCheckpoint`1+WorkCheckpointReactable[T].MustStop (UnityEngine.GameObject dupe, System.Single x) [0x0004c] in <628a95ce249f4b59997b3b2d4f17a15f>:0
  at PeterHan.QueueForSinks.WorkCheckpoint`1+WorkCheckpointReactable[T].InternalCanBegin (UnityEngine.GameObject newReactor, Navigator+ActiveTransition transition) [0x00045] in <628a95ce249f4b59997b3b2d4f17a15f>:0
  at Reactable.CanBegin (UnityEngine.GameObject reactor, Navigator+ActiveTransition transition) [0x000a5] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at ReactionMonitor+Instance.TryReact (Reactable reactable, System.Single clockTime, Navigator+ActiveTransition transition) [0x00037] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at ReactionMonitor+Instance.PollForReactables (Navigator+ActiveTransition transition) [0x000c9] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at ReactableTransitionLayer.BeginTransition (Navigator navigator, Navigator+ActiveTransition transition) [0x00014] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at TransitionDriver.BeginTransition (Navigator navigator, Navigator+ActiveTransition transition) [0x00032] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at TransitionDriver.BeginTransition (Navigator navigator, NavGrid+Transition transition, System.Single defaultSpeed) [0x00013] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at Navigator.BeginTransition (NavGrid+Transition transition) [0x0002b] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at Navigator.AdvancePath (System.Boolean trigger_advance) [0x00119] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at Navigator.GoTo (KMonoBehaviour target, CellOffset[] offsets, NavTactic tactic) [0x00068] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at Navigator.GoTo (System.Int32 cell, CellOffset[] offsets) [0x00024] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at GameStateMachine`4+State+<>c__DisplayClass123_0[StateMachineType,StateMachineInstanceType,MasterType,DefType].<MoveTo>b__0 (StateMachineInstanceType smi) [0x00034] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].ExecuteActions (StateMachine`4+State[StateMachineType,StateMachineInstanceType,MasterType,DefType] state, System.Collections.Generic.List`1[T] actions) [0x00048] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0

  at UnityEngine.Debug.LogError (System.Object message, UnityEngine.Object context) [0x00000] in <44795fdd2f534a10b35fa7129542a129>:0
  at Debug.LogError (System.Object obj, UnityEngine.Object context) [0x00000] in <6b448ec81ef2491a86c041b11073e018>:0
  at DebugUtil.LogErrorArgs (UnityEngine.Object context, System.Object[] objs) [0x00000] in <6b448ec81ef2491a86c041b11073e018>:0
  at DebugUtil.LogException (UnityEngine.Object context, System.String errorMessage, System.Exception e) [0x00000] in <6b448ec81ef2491a86c041b11073e018>:0
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].ExecuteActions (StateMachine`4+State[StateMachineType,StateMachineInstanceType,MasterType,DefType] state, System.Collections.Generic.List`1[T] actions) [0x00000] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].PushState (StateMachine+BaseState state) [0x00000] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].GoTo (StateMachine+BaseState base_state) [0x00000] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at IdleChore+States.<InitializeStates>b__6_4 (IdleChore+StatesInstance smi) [0x00000] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at GameStateMachine`4+State+<>c__DisplayClass99_0[StateMachineType,StateMachineInstanceType,MasterType,DefType].<ToggleScheduleCallback>b__2 (System.Object smi_data) [0x00000] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at Scheduler.Update () [0x00000] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
  at GameScheduler.Update () [0x00000] in <4e37f4deac7d4b7e83804005f8ea3c4d>:0
Build: U53-642443-S
```

## Output Log (from the 2nd video)
```
Exception in: (Minion).MoveToSafetyChore+States.root.move.MoveTo() 
System.TypeLoadException: Could not resolve type with token 01000036 from typeref (expected class 'Worker' in assembly 'Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null')
  at PeterHan.QueueForSinks.WorkCheckpoint`1+WorkCheckpointReactable[T].MustStop (UnityEngine.GameObject dupe, System.Single x) [0x0004c] in <628a95ce249f4b59997b3b2d4f17a15f>:0 
  at PeterHan.QueueForSinks.WorkCheckpoint`1+WorkCheckpointReactable[T].InternalCanBegin (UnityEngine.GameObject newReactor, Navigator+ActiveTransition transition) [0x00045] in <628a95ce249f4b59997b3b2d4f17a15f>:0 
  at Reactable.CanBegin (UnityEngine.GameObject reactor, Navigator+ActiveTransition transition) [0x000a5] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at ReactionMonitor+Instance.TryReact (Reactable reactable, System.Single clockTime, Navigator+ActiveTransition transition) [0x00037] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at ReactionMonitor+Instance.PollForReactables (Navigator+ActiveTransition transition) [0x000c9] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at ReactableTransitionLayer.BeginTransition (Navigator navigator, Navigator+ActiveTransition transition) [0x00014] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at TransitionDriver.BeginTransition (Navigator navigator, Navigator+ActiveTransition transition) [0x00032] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at TransitionDriver.BeginTransition (Navigator navigator, NavGrid+Transition transition, System.Single defaultSpeed) [0x00013] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at Navigator.BeginTransition (NavGrid+Transition transition) [0x0002b] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at Navigator.AdvancePath (System.Boolean trigger_advance) [0x00119] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at Navigator.GoTo (KMonoBehaviour target, CellOffset[] offsets, NavTactic tactic) [0x00068] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at Navigator.GoTo (System.Int32 cell, CellOffset[] offsets) [0x00024] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at GameStateMachine`4+State+<>c__DisplayClass123_0[StateMachineType,StateMachineInstanceType,MasterType,DefType].<MoveTo>b__0 (StateMachineInstanceType smi) [0x00034] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].ExecuteActions (StateMachine`4+State[StateMachineType,StateMachineInstanceType,MasterType,DefType] state, System.Collections.Generic.List`1[T] actions) [0x00048] in <9a9649e5bd67484e9df8ebd174a7adae>:0 

  at UnityEngine.Debug.LogError (System.Object message, UnityEngine.Object context) [0x00000] in <44795fdd2f534a10b35fa7129542a129>:0 
  at Debug.LogError (System.Object obj, UnityEngine.Object context) [0x00000] in <c77f460c85e646a99e24b9c78b66c579>:0 
  at DebugUtil.LogErrorArgs (UnityEngine.Object context, System.Object[] objs) [0x00000] in <c77f460c85e646a99e24b9c78b66c579>:0 
  at DebugUtil.LogException (UnityEngine.Object context, System.String errorMessage, System.Exception e) [0x00000] in <c77f460c85e646a99e24b9c78b66c579>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].ExecuteActions (StateMachine`4+State[StateMachineType,StateMachineInstanceType,MasterType,DefType] state, System.Collections.Generic.List`1[T] actions) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].PushState (StateMachine+BaseState state) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].GoTo (StateMachine+BaseState base_state) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine+Instance.StartSM () [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].StartSM () [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StandardChoreBase.Begin (Chore+Precondition+Context context) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at ChoreDriver+StatesInstance.BeginChore () [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at ChoreDriver+States+<>c.<InitializeStates>b__5_2 (ChoreDriver+StatesInstance smi) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].ExecuteActions (StateMachine`4+State[StateMachineType,StateMachineInstanceType,MasterType,DefType] state, System.Collections.Generic.List`1[T] actions) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].PushState (StateMachine+BaseState state) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].GoTo (StateMachine+BaseState base_state) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+Parameter`1+Transition[StateMachineType,StateMachineInstanceType,MasterType,DefType,ParameterType].Evaluate (StateMachine+Instance smi) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+Parameter`1+Context[StateMachineType,StateMachineInstanceType,MasterType,DefType,ParameterType].Set (ParameterType value, StateMachineInstanceType smi, System.Boolean silenceEvents) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at StateMachine`4+Parameter`1[StateMachineType,StateMachineInstanceType,MasterType,DefType,ParameterType].Set (ParameterType value, StateMachineInstanceType smi, System.Boolean silenceEvents) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at ChoreDriver.SetChore (Chore+Precondition+Context context) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at Brain.UpdateChores () [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at Brain.UpdateBrain () [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at MinionBrain.UpdateBrain () [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at BrainScheduler+BrainGroup.RenderEveryTick (System.Single dt) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at BrainScheduler.RenderEveryTick (System.Single dt) [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
  at SimAndRenderScheduler+RenderEveryTickUpdater.Update (IRenderEveryTick updater, System.Single dt) [0x00000] in <c77f460c85e646a99e24b9c78b66c579>:0 
  at UpdateBucketWithUpdater`1[DataType].Update (System.Single dt) [0x00000] in <c77f460c85e646a99e24b9c78b66c579>:0 
  at StateMachineUpdater+BucketGroup.AdvanceOneSubTick (System.Single dt) [0x00000] in <c77f460c85e646a99e24b9c78b66c579>:0 
  at StateMachineUpdater.RenderEveryTick (System.Single dt) [0x00000] in <c77f460c85e646a99e24b9c78b66c579>:0 
  at Game.LateUpdate () [0x00000] in <9a9649e5bd67484e9df8ebd174a7adae>:0 
Build: U53-642695-S
```

**Comment by mallardduck** [comment]
I was able to test with a local dev build and it seems to work. I think it just needs to be rebuilt with the new ONI libraries and it'll start working again.

**Comment by peterhaneve** [comment]
This issue has been fixed in version 3.14.0.0. It is likely that you will need Mod Updater to propagate the change.
